pelicun/tools/DL_calculation.py

```diff
@@ -389,6 +390,25 @@ def run_pelicun(
 
                 return 0
 
+            # look for possibly specified assessment options
+            try:
+                assessment_options = config['Applications']['DL']['ApplicationData'][
+                    'Options'
+                ]
+            except KeyError:
+                assessment_options = None
+
+            if assessment_options:
+                # extend options defined via the auto-population script to
+                # include those in the original `config`
+                config_ap['Applications']['DL']['ApplicationData'].pop('Options')
+                update_vals(
+                    config_ap['DL']['Options'],
```

Comment:
I am not sure I understand what you are doing here:
- First, you pop the `Options` from the `ApplicationData` part of the `config_ap`. Why?
- Then you update the `Options` under `DL` in the `config_ap` with the settings loaded from the `ApplicationData` part of the original `config`, not the `config_ap`.

If it was `config_ap` everywhere, it would make more sense. But now it is quite confusing with these two different config variables and the two different locations for `Options` in each.

Please help me understand.


Wouldn't we need an 'Applications' and a in there? Like so: `config_ap['Applications']['DL']['ApplicationData']['Options']

Reply:
- The `config` variable stores the `DL_calculation.py` input file, defined as a JSON file.
- There are two options for this file, it can either define a `DL` section or not. In the latter case, auto-population is required.
- The correct location to place the seed so that it is properly propagated to the Assessment object is `DL/Options/Seed`, but when auto-population is required, there is no `DL` section.
- To get around that, I introduced an additional placeholder for assessment options for the case of auto-population: `Applications/DL/ApplicationData/Options`. Input files meant for auto-population that need to specify options can have them defined there. When the auto-populated config file is generated, it maintains that data, but we have to move them to the right location, namely `DL/Options`. It is possible, however, for the autopopulation script (which we have no control over) to be such that it stores values there. That's why I used the `update_vals` function from `base`, previously used to merge user-defined and default settings, to merge any auto-populated `DL/Options` in `config_ap` with those coming from `Applications/DL/ApplicationData/Options`. For example, if a seed value is provided both in the input file and from the auto-population script, the one from the auto-population script will take precedence.